PokeDemo/Controllers/HomeViewController.swift

```diff
@@ -0,0 +1,91 @@
+//
+//  ViewController.swift
+//  PokeDemo
+//
+//  Created by Rudr Bansal on 05/07/22.
+//
+// MARK: - Navigation
+
+import UIKit
+
+class HomeViewController: BaseViewController {
+    
+    // MARK: - IBOutlets
+    
+    @IBOutlet weak var pokemonTableView: UITableView!
+    
+    // MARK: - Properties
+    
+    // MARK: - Private
+
+    private let presenter = HomeViewPresenter(service: Service.shared)
+    private var pokemonList: [Pokemon]? = [Pokemon]()
+    
+    // MARK: - View Life Cycle
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        pokemonTableView.register(UINib(nibName: PokemonTableViewCell.cellIdentifier(), bundle: nil), forCellReuseIdentifier: PokemonTableViewCell.cellIdentifier())
+        presenter.setViewDelegate(delegate: self)
+        presenter.getPokemonList()
+    }
+}
+
+// MARK: - Extensions
+
+extension HomeViewController: UITableViewDataSource, UITableViewDelegate {
+    
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return pokemonList?.count ?? 0
+    }
+    
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let cell = tableView.dequeueReusableCell(withIdentifier: PokemonTableViewCell.cellIdentifier()) as! PokemonTableViewCell
+        let pokemon = pokemonList![indexPath.row]
+        cell.setupData(data: pokemon)
+        presenter.getPokemonInfo(pokemon: pokemon) { image in
```

Comment:
Since architecture is MVP. view notifies to presenter instead of giving command. Here `getPokemonInfo` behaves as view is giving a command to presenter to do something. Rather view must use notification sort of function names. Such as `didSetupCell(with: pokemon)`

Consider view should not know what presenter is going to do with this information. View's job is to provide status to presenter that something happened at view and presenter will use these methods, to download image or do something which is not related to view. Thus it separates the concerns to specific layer of the architecture. Also, does not bleed out the information to different layers

Reply:
So if you use the simple method didSetupCell, you might need to pass some unique cell info, such as row or something, so that when presenter downloads the image it can call a separate function to view saying, `view.displayImage(image, at: indexRow)`

Reply:
Done in 6f1d95bb7ca89332952ecd23599b4c8834939145